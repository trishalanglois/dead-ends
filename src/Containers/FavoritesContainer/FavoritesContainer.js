import React from 'react';
import './FavoritesContainer.scss';
import LocationCard from '../LocationCard/LocationCard';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

export const FavoritesContainer = ({ favoriteLocations }) => {
  const showLocations = favoriteLocations.map(spookyLocation => {
    return (
      <>
        <LocationCard
          key={Math.random() * Math.random()}
          location={spookyLocation}
        />
      </>
    )
  })

  return (
    <main className="locations-container">
      <div className="locations-container-subheader">
        <h2 className="location-subheader">Favorite Locations</h2>
        <Link to={'/locations'}>
          <h3 className="favorites-button">Back</h3>
        </Link>
      </div>
      <div className="main-section">
        <aside className="locations-cards-container">
          {showLocations}
        </aside>
        <aside className="map-container">
          //render MapContainer component
        </aside>
      </div>
    </main>
  )
}

export const mapStateToProps = state => ({
  favoriteLocations: state.favorites
})


export default connect (mapStateToProps)(FavoritesContainer);
