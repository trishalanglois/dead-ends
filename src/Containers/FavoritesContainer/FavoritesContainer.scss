@import '../../scss/variables';

.no-locations-error {
  font-family: $accent-font;
  color: $light-grey;
  font-size: 2em;
  font-weight: 200;
  height: 100vh;
  padding-top: 5em;
  margin-top: -1em;
}
