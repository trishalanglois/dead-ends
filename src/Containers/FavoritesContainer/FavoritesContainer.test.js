import React from 'react';
import { FavoritesContainer, mapStateToProps } from './FavoritesContainer';
import { shallow } from 'enzyme';

describe('FavoritesContainer', () => {
  it('should match the snapshot when there are no favorite locations', () => {
    let mockProps = {
      favoriteLocations: []
    };
    let wrapper = shallow(<FavoritesContainer
      favoriteLocations={mockProps.favoriteLocations}
    />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should match the snapshot when there are favorite locations', () => {
    let mockProps = {
      favoriteLocations: [
        {
          "city": "Ada",
          "country": "United States",
          "description": "BOO!",
          "location": "Ada Cemetery",
          "state": "Michigan",
          "state_abbrev": "MI",
          "longitude": -85.50489309999999,
          "latitude": 42.9621061,
          "city_longitude": -85.4954803,
          "city_latitude": 42.960727
        }
      ]
    };
    let wrapper = shallow(<FavoritesContainer
      favoriteLocations={mockProps.favoriteLocations}
    />)

    expect(wrapper).toMatchSnapshot()
  })

  describe('mapStateToProps', () => {
    it('should return an object with the favorite locations array', () => {
      const mockState = {
        favorites: [
          {
            "city": "Ada",
            "country": "United States",
            "description": "BOO!",
            "location": "Ada Cemetery",
            "state": "Michigan",
            "state_abbrev": "MI",
            "longitude": -85.50489309999999,
            "latitude": 42.9621061,
            "city_longitude": -85.4954803,
            "city_latitude": 42.960727
          },
          {
            "city": "Addison",
            "country": "United States",
            "description": "Spooky stuff.",
            "location": "North Adams Rd.",
            "state": "Michigan",
            "state_abbrev": "MI",
            "longitude": -84.3818434,
            "latitude": 41.9714248,
            "city_longitude": -84.3471684,
            "city_latitude": 41.986434
          }
        ]
      };
      const expected = {
        favoriteLocations: [{
          "city": "Ada",
          "country": "United States",
          "description": "BOO!",
          "location": "Ada Cemetery",
          "state": "Michigan",
          "state_abbrev": "MI",
          "longitude": -85.50489309999999,
          "latitude": 42.9621061,
          "city_longitude": -85.4954803,
          "city_latitude": 42.960727
        },
        {
          "city": "Addison",
          "country": "United States",
          "description": "Spooky stuff.",
          "location": "North Adams Rd.",
          "state": "Michigan",
          "state_abbrev": "MI",
          "longitude": -84.3818434,
          "latitude": 41.9714248,
          "city_longitude": -84.3471684,
          "city_latitude": 41.986434
        }]
      }

      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    })
  })
})
