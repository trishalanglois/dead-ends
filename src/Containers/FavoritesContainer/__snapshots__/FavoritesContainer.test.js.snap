// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesContainer should match the snapshot when there are favorite locations 1`] = `
<main
  className="locations-container"
>
  <div
    className="locations-container-subheader"
  >
    <h2
      className="location-subheader"
    >
      Favorite Spooky Locations
    </h2>
    <Link
      to="/locations"
    >
      <h3
        className="favorites-button"
      >
        Back
      </h3>
    </Link>
  </div>
  <div
    className="main-section"
  >
    <aside
      className="locations-cards-container"
    >
      <Connect(LocationCard)
        cardLocation={
          Object {
            "city": "Ada",
            "city_latitude": 42.960727,
            "city_longitude": -85.4954803,
            "country": "United States",
            "description": "BOO!",
            "latitude": 42.9621061,
            "location": "Ada Cemetery",
            "longitude": -85.50489309999999,
            "state": "Michigan",
            "state_abbrev": "MI",
          }
        }
        key="0.020955581196682962"
      />
    </aside>
  </div>
</main>
`;

exports[`FavoritesContainer should match the snapshot when there are no favorite locations 1`] = `
<main
  className="locations-container"
>
  <div
    className="locations-container-subheader"
  >
    <h2
      className="location-subheader"
    >
      Favorite Spooky Locations
    </h2>
    <Link
      to="/locations"
    >
      <h3
        className="favorites-button"
      >
        Back
      </h3>
    </Link>
  </div>
  <div
    className="main-section"
  >
    <aside
      className="locations-cards-container"
    >
      <h3
        className="no-locations-error"
      >
        Add favorite locations to view them here.
      </h3>
    </aside>
  </div>
</main>
`;
